### ISSUE TITLE
SyntaxError: Named export 'SfError' not found

### ISSUE THREAD

**Issue by JoniJarvinen** [jayree/sfdx-plugin-manifest#885]
Hi!

I am getting error:
`SyntaxError: Named export 'SfError' not found. The requested module '@salesforce/core' is a CommonJS module, which may not support all module.exports as named exports.
    CommonJS modules can always be imported via the default export, for example using:

    import pkg from '@salesforce/core';
    const { SfError, Lifecycle } = pkg;`

while trying to perform a manifest git diff. This started happening in the past week or so.

Not much to work with but there could have been an update in some of the dependencies.

**Comment by jayree** [comment]
@JoniJarvinen could you please try again with the latest version of the plugin and share the output of 
`sfdx plugins --core` 
thanks.

**Comment by JoniJarvinen** [comment]
Hello,

`sfdx-cli: Updating plugins... | [2/4] Fetching packages...
warning "@salesforce/lwc-dev-server > @rollup/plugin-alias@3.1.9" has unmet peer dependency "rollup@^1.20.0||^2.0.0".
warning "@salesforce/lwc-dev-server > request-promise-native@1.0.9" has unmet peer dependency "request@^2.34".
warning "@salesforce/lwc-dev-server > request-promise-native > request-promise-core@1.1.4" has unmet peer dependency "request@^2.34".
warning "@jayree/sfdx-plugin-manifest > @salesforce/sf-plugins-core > @oclif/core > ts-node@10.9.1" has unmet peer dependency "@types/node@*".
sfdx-cli: Updating plugins... done
aa> sfdx jayree:manifest:git:diff local_a..local_b --source-dir=./package/main/default/ --output-dir=./some_dir
    SyntaxError: Named export 'SfError' not found. The requested module '@salesforce/core' is a CommonJS module, which may not support all module.exports as named exports.
    CommonJS modules can always be imported via the default export, for example using:

    import pkg from '@salesforce/core';
    const { SfError, Lifecycle } = pkg;

aa> sfdx plugins --core
@jayree/sfdx-plugin-manifest 3.1.21
@oclif/plugin-autocomplete 2.3.1 (core)
@oclif/plugin-commands 2.2.17 (core)
@oclif/plugin-help 5.2.11 (core)
@oclif/plugin-not-found 2.3.26 (core)
@oclif/plugin-plugins 3.1.6 (core)
@oclif/plugin-search 0.0.18 (core)
@oclif/plugin-update 3.1.22 (core)
@oclif/plugin-version 1.3.6 (core)
@oclif/plugin-warn-if-update-available 2.0.41 (core)
@oclif/plugin-which 2.2.22 (core)
@salesforce/lwc-dev-server 2.11.0
apex 2.3.4 (core)
auth 2.8.4 (core)
community 2.3.5 (core)
custom-metadata 2.1.30 (core)
data 2.4.5 (core)
deploy-retrieve 1.13.5 (core)
info 2.6.30 (core)
limits 2.3.23 (core)
org 2.9.19 (core)
packaging 1.20.1 (core)
schema 2.3.17 (core)
settings 1.4.17 (core)
sfdx-cli 7.209.6 (core)
signups 1.4.22 (core)
source 2.10.22 (core)
telemetry 2.2.1 (core)
templates 55.5.2 (core)
trust 2.4.27 (core)
user 2.3.20 (core)`

I haven't moved onto "sf" yet and using "sfdx" currently. Some SFDX release notes:
`IMPORTANT ANNOUNCEMENT, PLEASE READ: We no longer update sfdx (v7); the last stable version is 7.209.6. You can keep using sfdx (v7) if you want, but it won't contain new features or bug fixes. And we no longer update this page.

To continue getting weekly CLI updates, you must move to sf (v2).  It's easy: simply uninstall sfdx and then install sf. See the new Move from sfdx (v7) to sf (v2) (https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_move_to_sf_v2.htm) section of the Setup Guide for details, including how to update your continuous integration (CI) scripts. We also updated the entire Setup Guide to assume you're using sf (v2) and the sf-style CLI commands and configuration.

The sf (v2) Release Notes are here (https://github.com/forcedotcom/cli/tree/main/releasenotes/sfdx/../README.md); start checking that page for information about our weekly Salesforce CLI releases.
For context and information about this change, see this blog post (https://developer.salesforce.com/blogs/2023/07/salesforce-cli-sf-v2-is-here).`

Should I try to move away from SFDX into sf v2?

**Comment by jayree** [comment]
@JoniJarvinen yes, please migrate to `sf`. My first guess: Since `sfdx-cli` is no longer maintained, the newer dependency updates just messed up. But I'll see what I can find out.

**Comment by JoniJarvinen** [comment]
@jayree Thanks for the responses. By upgrading to sf v2 everything started working as before. Maybe a note in the readme would do. I tried downgrading versions until June but that did not help. So without much investigation in my case the older sfdx dependecies aren't compatible anymore, unless you have a quick fix like conditionally importing them depending on versions.
